src/env.h

```diff
@@ -263,6 +265,9 @@ class Environment {
     enum Fields {
       // Set this to not zero if the init hook should be called.
       kCallInitHook,
+      // Set this to not zero if async wrap should be enabled. This is
+      // automatically set when SetupHooks is called.
+      kEnabled,
```

Comment:
I know you know this, but to reiterate. The point was that `before`/`after` don't run unless `init` runs, so enabling `init` implies enabling the others. I can see your confusion though, reading though the code the implicit constraint on what callbacks are called isn't apparent.